From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: corymhall <43035978+corymhall@users.noreply.github.com>
Date: Mon, 16 Jun 2025 08:26:37 -0400
Subject: [PATCH] Handle cases where region is not a string

Related to
https://github.com/pulumi/pulumi-terraform-bridge/issues/2281. TF sets
`region` in an interceptor which means for Pulumi `region` could either
not exist at all in `RawState` or it could be `nil`.

diff --git a/internal/provider/sdkv2/provider.go b/internal/provider/sdkv2/provider.go
index f6b44b7c2b..79fdfb07e0 100644
--- a/internal/provider/sdkv2/provider.go
+++ b/internal/provider/sdkv2/provider.go
@@ -745,7 +745,9 @@ func (p *sdkProvider) initialize(ctx context.Context) (map[string]conns.ServiceP
 
 					if isRegionOverrideEnabled && getAttribute != nil {
 						if region, ok := getAttribute(names.AttrRegion); ok {
-							overrideRegion = region.(string)
+							if val, ok := region.(string); ok {
+								overrideRegion = val
+							}
 						}
 					}
 
